### PR TITLE
ArcSight Module Broken (Invalid Type), Fixed

### PR DESCRIPTION
**Pull request by jsvd**
* index arcsight data with document type `_doc` so it works on ES 8.x
* include Logstash version in index names to allow different document types to facilitate upgrading
* remove dangling `var.elasticsearch.index_suffix` setting

For reviewers:
1. Install Kibana and ES 6.x
2. Run Logstash 7.x Arcsight Module like so:

```
❯ bin/logstash --modules arcsight --setup -M "arcsight.var.inputs=smartconnector"  -M "arcsight.var.input.smartconnector.port=5010" -M "arcsight.var.elasticsearch.hosts=localhost:9200"  -M "arcsight.var.kibana.host=localhost:5601" -M "arcsight.var.kibana.ssl.enabled=false"
```

3. Ingest data like so:

```
echo "CEF:0|FooBar|Web Gateway|1.2.3.45.67|200|Success|2|rt=Sep 07 2018 14:50:39 cat=Access Log dst=1.1.1.1 dhost=[foo.example.com](http://foo.example.com/) suser=redacted src=2.2.2.2 requestMethod=POST request='https://foo.example.com/bar/bingo/1' requestClientApplication='Foo-Bar/2018.1.7; Email:[user@example.com](mailto:user@example.com); Guid:test=' cs1= cs1Label=Foo Bar\r\n" | nc -v localhost 5010
```

4. Stop Logstash 6.x
5. Stop and Upgrade ES and Kibana to 7.x
6. Use Logstash 8.x with this PR to ingest data like steps 2 and 3
7. Observe that the ES cluster will ingest data correctly, with two indices:
  a. "arcsight-YYYY.MM.dd" for the data ingested in step 3
  b. "arcsight-8.4.0-YYYY.MM.dd" for data ingested in step 6
8. Observe that document type for the single document in "arcsight-YYYY.MM.dd" is "syslog"
9. Observe that document type for the single document in "arcsight-8.4.0-YYYY.MM.dd" is "_doc"

replaces #11504

### CODE REVIEW

**Comment by jsvd** [comment]
Regarding upgrades of Logstash, there are a few concerns:

For users on Elasticsearch 6.x the arcsight indices have type syslog. This PR would break the current daily index as 1 index can't have multiple types, but this can be disregarded as Logstash 8.0.0 doesn't support ES 6.x.

For users on Elasticsearch 7.x the issue is similar: upgrading from Logstash 7.x to Logstash 8.x with this PR would attempt to write a second type to the current daily index, blocking indexing. Documentation can be written to guide users to reindex and delete that daily index before upgrading Logstash.

For users on Elasticsearch 8.0 the current situation is broken as ES8 doesn't accept the syslog type on indices. So this PR is clearly beneficial. 

To summarize, the main concern is for ES7 users that upgrade from Logstash 7 to 8.

**Comment by jsvd** [comment]
> To summarize, the main concern is for ES7 users that upgrade from Logstash 7 to 8.

A way to avoid conflicts between the indices with different types is to name the indices going forward with an extra element in the prefix such as the Logstash major.minor version: "arcsight-v8.1-*"

**Comment by MikePaquette** [comment]
Hi @jsvd any update on when this might be merged?  We met with a customer yesterday who had been tracking this.  Thx!

**Comment by jsvd** [comment]
@MikePaquette while the code change is small, some extra work is needed to add unit testing and to confirm this works in different upgrade scenarios (Logstash first, Elasticsearch first, etc).
I'll try to find some time in the next couple of weeks to move this along.

**Review by kaisecheng — APPROVED** [review]
Test locally and work as expected. LGTM

**Comment by jsvd** [comment]
@MikePaquette the fix has been merged and should ship in the next minor release.
